### PR TITLE
[rush] Rename subspace pnpmfile to .pnpmfile.cjs from .pnpmfile-subspace.cjs

### PR DESCRIPTION
**Pull request by william2958**
## Summary

Design note: https://github.com/microsoft/rushstack/issues/4720

This MR renames the subspace level pnpmfile to ".pnpmfile.cjs" from ".pnpmfile-subspace.cjs", as we are no longer allowing a rush level ".pnpmfile.cjs" file to exist.

This MR also checks if a ".npmrc" file exists globally, and throws an error if it does, prompting the user to use the ".npmrc-global" file instead. It also throws an error if the ".pnpmfile.cjs" file exists at the rush config level.

## Impacted documentation
This affects the documentation listed here: https://rushjs.io/pages/advanced/subspaces/

### CODE REVIEW

File: common/changes/@microsoft/rush/will-subspaces-patch-26_2024-05-20-22-22.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft/rush",
+      "comment": "Rename subspace level pnpmfile from .pnpmfile-subspace.cjs to .pnpmfile",
```
Comment (octogonz) [inline common/changes/@microsoft/rush/will-subspaces-patch-26_2024-05-20-22-22.json:5]:
```suggestion
      "comment": "Rename the subspace level lockfile from `.pnpmfile-subspace.cjs` to `.pnpmfile.cjs`. This is a breaking change for the experimental feature.",
```

**Comment by octogonz** [comment]
image

This certainly requires doc updates

File: libraries/rush-lib/src/api/Subspace.ts [hunk 2]
```diff
@@ -134,11 +134,6 @@ export class Subspace {
               `The project level configuration file ${project.projectFolder}/.npmrc is no longer valid. Please use a ${subspaceConfigFolder}/.npmrc file instead.`
             );
           }
-          if (FileSystem.exists(`${project.projectFolder}/.pnpmfile.cjs`)) {
-            throw new Error(
-              `The project level configuration file ${project.projectFolder}/.pnpmfile.cjs is no longer valid. Please use a ${subspaceConfigFolder}/.pnpmfile-subspace.cjs file instead.`
```
Comment (octogonz) [inline libraries/rush-lib/src/api/Subspace.ts:139]: @william2958 Why are we removing this validation?

Seems like we should keep this check, but change the error message to recommend `${subspaceConfigFolder}/.pnpmfile.cjs`

Comment (william2958) [reply libraries/rush-lib/src/api/Subspace.ts:139]: Added back

**Review by octogonz — APPROVED** [review]

**Pull request merged**